src/go/k8s/controllers/redpanda/cluster_controller.go

```diff
@@ -112,194 +112,129 @@ func (r *ClusterReconciler) Reconcile(
 	log.Info("Starting reconcile loop")
 	defer log.Info("Finished reconcile loop")
 
-	var redpandaCluster vectorizedv1alpha1.Cluster
-	crb := resources.NewClusterRoleBinding(r.Client, &redpandaCluster, r.Scheme, log)
-	if err := r.Get(ctx, req.NamespacedName, &redpandaCluster); err != nil {
+	var vectorizedCluster vectorizedv1alpha1.Cluster
+	ar := newAttachedResources(ctx, r, log, &vectorizedCluster)
+	if err := r.Get(ctx, req.NamespacedName, &vectorizedCluster); err != nil {
 		// we'll ignore not-found errors, since they can't be fixed by an immediate
 		// requeue (we'll need to wait for a new notification), and we can get them
 		// on deleted requests.
 		if apierrors.IsNotFound(err) {
-			if removeError := crb.RemoveSubject(ctx, req.NamespacedName); removeError != nil {
+			if removeError := ar.getClusterRoleBinding().RemoveSubject(ctx, req.NamespacedName); removeError != nil {
 				return ctrl.Result{}, fmt.Errorf("unable to remove subject in ClusterroleBinding: %w", removeError)
 			}
 			return ctrl.Result{}, nil
 		}
 		return ctrl.Result{}, fmt.Errorf("unable to retrieve Cluster resource: %w", err)
 	}
-	// if the cluster is being deleted, delete finalizers
-	if !redpandaCluster.GetDeletionTimestamp().IsZero() {
-		return r.handleClusterDeletion(ctx, &redpandaCluster, log)
+
+	isManaged := isRedpandaClusterManaged(log, &vectorizedCluster) && isRedpandaClusterVersionManaged(log, &vectorizedCluster, r.RestrictToRedpandaVersion)
```

Comment:
This being switched? The current logic is to first check if things are being deleted before we check if we are being managed. This will allow deletion to complete once it has been started, and then determine if we are managed or not. That order is important i think and probably is relied upon, what are your thoughts on this?

Reply:
My thought was, if this is being tagged as unmanaged by this controller, then why does this controller hold the finalizer? Seems like it shouldn't.

Reply:
In principle I most definitely agree. So essentially if we set to not be managed, and then delete, it will not be deleted at all. That's fine with me but once logic like this is in place, its usually heavily expected, my only concern is that the expectation.

Reply:
I think I'm going to keep this and, as @RafalKorepta pointed out, just call it out in the release notes.